packages/experimental/src/experimental-list/collapsible-list/index.tsx

```diff
@@ -36,6 +43,41 @@ function getContainerHeight( collapseContainer: HTMLDivElement | null ) {
 	return containerHeight;
 }
 
+/**
+ * This functions returns a new list of shown children depending on the new children updates.
+ * If one is removed, it will remove it from the show array.
+ * If one is added, it will add it back to the shown list, making use of the new children list to keep order.
+ *
+ * @param {Array.<import('react').ReactElement>} currentChildren a list of the current children.
+ * @param {Array.<import('react').ReactElement>} currentShownChildren a list of the current shown children.
+ * @param {Array.<import('react').ReactElement>} newChildren a list of the new children.
+ * @return {Array.<import('react').ReactElement>} new list of children that should be shown.
+ */
+function getUpdatedShownChildren(
+	currentChildren: React.ReactElement[],
+	currentShownChildren: React.ReactElement[],
+	newChildren: React.ReactElement[]
+): React.ReactElement[] {
```

Comment:
Is this covered by tests?

Reply:
No, I tried writing some tests of the actual staggered transition, but jest didn't like the `useFakeTimers` very much.
I will add tests for this function.